sdk/communication/azure-communication-callingserver/tests/test_call_connection.py

```diff
@@ -0,0 +1,1070 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import unittest
+import utils._test_mock_utils as _mock_utils
+import utils._test_constants as _test_constants
+
+from parameterized import parameterized
+from typing import List
+
+from azure.communication.callingserver import (
+    AudioRoutingMode,
+    CommunicationIdentifier,
+    PhoneNumberIdentifier
+    )
+
+from utils._unit_test_utils import CallConnectionUnitTestUtils
+
+class TestCallConnection(unittest.TestCase):
+
+    @parameterized.expand(CallConnectionUnitTestUtils.data_source_test_get_call())
+    def test_get_call_succeed(
+        self,
+        test_name, # type: str
+        call_connection_id, # type: str
+        use_managed_identity = False, # type: bool
+        ):
+
+        call_connection = _mock_utils.create_mock_call_connection(
+            call_connection_id,
+            status_code=200,
+            payload=_test_constants.GetCallResponsePayload,
+            use_managed_identity=use_managed_identity
+            )
+
+        result = call_connection.get_call()
+        CallConnectionUnitTestUtils.verify_get_call_result(result)
+
+    @parameterized.expand(CallConnectionUnitTestUtils.data_source_test_get_call())
+    def test_get_call_failed(
+        self,
+        test_name, # type: str
+        call_connection_id, # type: str
+        use_managed_identity = False, # type: bool
+        ):
+
+        call_connection = _mock_utils.create_mock_call_connection(
+            call_connection_id,
+            status_code=404,
+            payload=_test_constants.ErrorPayload,
+            use_managed_identity = use_managed_identity
+            )
+
+        raised = False
+        try:
+            call_connection.get_call()
+        except:
+            raised = True
+        assert raised == True
+
+    @parameterized.expand(CallConnectionUnitTestUtils.data_source_test_delete())
+    def test_delete_succeed(
+        self,
+        test_name, # type: str
+        call_connection_id, # type: str
+        use_managed_identity = False, # type: bool
+        ):
+
+        call_connection = _mock_utils.create_mock_call_connection(
+            call_connection_id,
+            status_code=202,
+            payload=_test_constants.GetCallResponsePayload,
+            use_managed_identity=use_managed_identity
+            )
+
+        call_connection.delete()
+        assert call_connection.call_connection_id == _test_constants.CALL_ID
```

Comment:
does this validate that the call was deleted? It also seems to be odd that you are validating that the value for call_connection_id, which you pass in, is equal to _test_constants.CALL_ID. do you need to pass in call_connection_id?

Reply:
It looks like we are just relying on delete not throwing an exception as validation that the connection was deleted.